jbmc/src/java_bytecode/replace_java_nondet.cpp

```diff
@@ -296,8 +295,9 @@ static goto_programt::targett check_and_replace_target(
     inserted_expr.set_nullable(
       instr_info.get_nullable_type() ==
       nondet_instruction_infot::is_nullablet::TRUE);
-    target_instruction->code = code_assignt(nondet_var, inserted_expr);
-    target_instruction->code.add_source_location() =
+    target_instruction->code_nonconst() =
+      code_assignt(nondet_var, inserted_expr);
+    target_instruction->code_nonconst().add_source_location() =
       target_instruction->source_location;
```

Comment:
Just like the above: it's an `ASSIGN` instruction already, don't we just need to replace lhs/rhs here?